lib/KThemePlugin.js

```diff
@@ -66,6 +67,12 @@ export default function KThemePlugin(Vue) {
         return this.$mediaType === 'print';
       },
     },
+    mounted() {
+      if (this.$el === this.$root.$el) {
```

Comment:
I just wonder if there is a problem with the order in which the mounted hooks are executed. Because a parent component finishes mounting when all its children have been mounted, so if, for some reason in the first render, some child component calls sendPoliteMessage in its mounted hook, in theory _mountLiveRegion would not have been called yet because it would be the last one mounted hook to be called on the first render.

btw I'm also not sure if calling sendPoliteMessage on a mounted hook is an appropriate use case, but I was also thinking about the use case of inserting the tooltips container div, and was wondering if the same strategy could be used for it.

Reply:
Thank you Alex, hm yes, I think that may happen, even though I haven't seen it yet amongst many other issues I experienced. I haven't implemented yet, but hoped this TODO item should make everything even more bulletproof, including the situation you are describing if I understand it right:

https://github.com/learningequality/kolibri-design-system/pull/687/files#diff-eacdfbe0cbeff51092f2072e0b0ba63206fed4c9341494075cd77c3667ce6382R72-R74.

How does that look to you? Any other ideas?

Reply:
The safest option would be to always check for the presence of the region in dom before sending the message but that sounded to me like potentially lots of queries and also am sure if it would cause some delays in delivering the message to the screen reader. That's why the idea about just catching error.

Reply:
Oh but as we already have these items in the window object, we can just check them. One Idea I had was to check and mount the live-region if needed on demand inside the `sendAssertiveMessage` and `sendPoliteMessage`, something just like `if (!window.politeLiveRegion) { _mountLiveRegion)() }`, and avoid mounting it in the root mounted hook.

But yes, that could potentially cause a little delay on the first call, but it would ensure that no message would be missed. We could meassure the performance impact of mounting it on demand, and see if it actually cause a delay issue.

What do you think?

Reply:
> One Idea I had was to check and mount the live-region if needed on demand inside the sendAssertiveMessage 

Yes, that's a check (and potential re-mount) that will be good  (even though my first thinking was rather using try/catch since it seems to be more robust than relying on window check since it would inform us any time something goes wrong?)

> and avoid mounting it in the root mounted hook

I don't follow exactly how that would help - one of the main objectives is to get live regions to DOM as soon as possible (but after it's more or less ready) and if we completely removed using the mounted hook, the mounting would need to wait until we call `'sendAssertiveMessage'` the first time. From the resource I saw, this is rather discouraged.

But if motivation for that suggestion is to ensure that no message would be missed as you say later, combining both these approaches should do, I believe?

Reply:
> My main thought about removing the check on the mounted hook was more about avoiding these comparison operations

I see. I thought it should hopefully not be such an issue since it's common to use `mounted` hooks in many components anyways, but definitely a valid concern since the condition is run many more times now than it would otherwise. In a sense, that's what I was trying to do in the first version that was just an event observing if a page has finished loading (however, there was some trouble recognizing that Vue parts are ready) Let me spend a while on some more experimentation and see if we can improve this aspect.